src/wp-includes/class-wp-image-editor-imagick.php

```diff
@@ -107,15 +107,30 @@ public static function supports_mime_type( $mime_type ) {
 		 * Here, we just say no if you are missing it and aren't loading a jpeg.
 		 */
 		if ( ! method_exists( 'Imagick', 'setIteratorIndex' ) && 'image/jpeg' !== $mime_type ) {
-				return false;
+			return false;
+		}
+
+		// Imagick::queryFormats is not performant, so cache the results.
+		$supports = wp_cache_get( 'imagick_supports' );
```

Comment:
We shouldn't use the object cache without a group. Not sure what would be most fitting for this use-case, but I think we should try to find one.